src/import/ImportRaw.cpp

```diff
@@ -288,31 +274,32 @@ void ImportRaw(const AudacityProject &project, wxWindow *parent, const wxString
 
 enum {
    ChoiceID = 9000,
-   PlayID
+   PlayID,
+   DetectID,
 };
 
 BEGIN_EVENT_TABLE(ImportRawDialog, wxDialogWrapper)
    EVT_BUTTON(wxID_OK, ImportRawDialog::OnOK)
    EVT_BUTTON(wxID_CANCEL, ImportRawDialog::OnCancel)
    EVT_BUTTON(PlayID, ImportRawDialog::OnPlay)
+   EVT_BUTTON(DetectID, ImportRawDialog::OnDetect)
    EVT_CHOICE(ChoiceID, ImportRawDialog::OnChoice)
 END_EVENT_TABLE()
 
-ImportRawDialog::ImportRawDialog(wxWindow * parent,
-                                 int encoding, unsigned channels,
-                                 int offset, double rate)
+ImportRawDialog::ImportRawDialog(wxWindow * parent, const wxString & fileName)
 :  wxDialogWrapper(parent, wxID_ANY, XO("Import Raw Data"),
             wxDefaultPosition, wxDefaultSize,
             wxDEFAULT_DIALOG_STYLE | wxRESIZE_BORDER),
-    mEncoding(encoding),
-    mChannels(channels),
-    mOffset(offset),
-    mRate(rate)
+   mFileName(fileName)
 {
-   wxASSERT(channels >= 1);
+   wxASSERT(mChannels >= 1);
```

Comment:
I see an assertion.  I look through the rest to find out what proves this assertion.  I see `FormatClassifier::GetResultChannels()` is the ultimate source of the value of mChannels.  I see to that the value (minus one) indexes into the choice control for mono or stereo.

So: (1) the condition should be stronger, that the value is 1 or 2, and (2) that should be stated as a postcondition (use `@post` in a Doxygen comment) on the method of the class FormatClassifier.

This is incidental to your changes -- you changed a variable name, and changed this line necessarily.  But let's clean up the assertion that exists.  Make it something more provable in terms of stated invariants of other classes.

Reply:
`FormatClassifier::GetResultChannels()` is not always source of `mChannels`. `mChannels` value is last channel import setting, with initial value of 1. So its value range is 1~16.